src/Sulu/Bundle/ResourceBundle/Resource/FilterListBuilder.php

```diff
@@ -0,0 +1,173 @@
+
+use Sulu\Bundle\ResourceBundle\Api\Condition;
+use Sulu\Bundle\ResourceBundle\Api\ConditionGroup;
+use Sulu\Bundle\ResourceBundle\Resource\Exception\ConditionFieldNotFound;
+use Sulu\Bundle\ResourceBundle\Resource\Exception\ConditionTypeMismatchException;
+use Sulu\Bundle\ResourceBundle\Resource\Exception\FilterNotFoundException;
+use Sulu\Component\Rest\ListBuilder\AbstractFieldDescriptor;
+use Sulu\Component\Rest\ListBuilder\ListBuilderInterface;
+use Sulu\Exception\FeatureNotImplementedException;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Component which triggers the generation of additional statements from the conditions of a filter
+ * and applies them to the list builder
+ *
+ * Class FilterListBuilder
+ * @package Sulu\Bundle\ResourceBundle\Resource
+ */
+class FilterListBuilder implements FilterListBuilderInterface
+{
+    /**
+     * @var RequestStack
+     */
+    protected $requestStack;
+
+    /**
+     * @var FilterManagerInterface
+     */
+    protected $manager;
+
+    /**
+     * @var ListBuilderInterface
+     */
+    protected $lb;
+
+    /**
+     * @param FilterManagerInterface $manager
+     * @param RequestStack           $requestStack
+     */
+    public function __construct(FilterManagerInterface $manager, RequestStack $requestStack)
+    {
+        $this->manager = $manager;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function applyFilterToList(ListBuilderInterface $lb)
+    {
+        $request = $this->requestStack->getCurrentRequest();
+        $locale = $request->getLocale();
+        $filterId = $request->get('filter');
+
+        $this->lb = $lb;
+
+        // when a filter is set
+        if ($filterId) {
+            $filter = $this->manager->findByIdAndLocale($filterId, $locale);
+
+            if (!$filter) {
+                throw new FilterNotFoundException($filterId);
+            }
+
+            foreach ($filter->getConditionGroups() as $conditionGroup) {
+                $this->processConditionGroup($conditionGroup, $filter->getConjunction());
+            }
+        }
+    }
+
+    /**
+     * Creates a conditions for a condition group
+     *
+     * @param ConditionGroup $conditionGroup
+     * @param string         $conjunction
+     *
+     * @throws ConditionFieldNotFound
+     * @throws FeatureNotImplementedException
+     */
+    protected function processConditionGroup(ConditionGroup $conditionGroup, $conjunction)
+    {
+        $condition = $conditionGroup->getConditions()[0];
+        $fieldDescriptor = $this->lb->getField($condition->getField());
+
+        if (!$fieldDescriptor) {
+            throw new ConditionFieldNotFound($condition->getField());
+        }
+
+        if (count($conditionGroup->getConditions()) === 1) {
+            $this->createCondition($condition, $fieldDescriptor, $conjunction);
+        } elseif (count($conditionGroup->getConditions()) === 2) {
```

Comment:
It just groups conditions but for now we don't support more than two conditions.